crates/orderbook/src/main.rs

```diff
@@ -591,12 +590,32 @@ async fn main() {
         Some(args.native_price_cache_max_update_size),
     );
 
-    let cow_subsidy = match args.cow_token_address {
-        Some(address) => Arc::new(CowSubsidyImpl::new(
-            ERC20::at(&web3, address),
-            args.cow_fee_factors.unwrap_or_default(),
-        )) as Arc<dyn CowSubsidy>,
-        None => Arc::new(FixedCowSubsidy(1.0)) as Arc<dyn CowSubsidy>,
+    let cow_token = match CowProtocolToken::deployed(&web3).await {
+        Err(DeployError::NotFound(_)) => None,
+        other => Some(other.unwrap()),
+    };
+    let cow_vtoken = match CowProtocolVirtualToken::deployed(&web3).await {
+        Err(DeployError::NotFound(_)) => None,
+        other => Some(other.unwrap()),
+    };
+    let cow_tokens = match (cow_token, cow_vtoken) {
+        (None, None) => None,
+        (Some(token), Some(vtoken)) => Some((token, vtoken)),
+        _ => panic!("should either have both cow token contracts or none"),
+    };
+    let cow_subsidy = match cow_tokens {
```

Comment:
Personally, I would have just used
```suggestion
    let cow_subsidy = match cow_token.zip(cow_vtoken) {
```

And not handled the case where only one of the tokens exist (its not like we can ever hit this codepath because of how we do contract deployments).

On the other hand - its already in there and this feels more "correct".